nibabel/arrayproxy.py

```diff
@@ -203,23 +203,20 @@ def _should_keep_file_open(self, file_like, keep_file_open):
         -------
 
         The value of ``keep_file_open`` that will be used by this
-        ``ArrayProxy``.
+        ``ArrayProxy``, and passed through to ``ImageOpener`` instances.
         """
         if keep_file_open is None:
             keep_file_open = KEEP_FILE_OPEN_DEFAULT
-        # if keep_file_open is True/False, we do what the user wants us to do
-        if isinstance(keep_file_open, bool):
-            return keep_file_open
-        if keep_file_open != 'auto':
+        if keep_file_open not in ('auto', True, False):
             raise ValueError('keep_file_open should be one of {None, '
                              '\'auto\', True, False}')
-
         # file_like is a handle - keep_file_open is irrelevant
         if hasattr(file_like, 'read') and hasattr(file_like, 'seek'):
             return False
-        # Otherwise, if file_like is gzipped, and we have_indexed_gzip, we set
-        # keep_file_open to True, else we set it to False
-        return HAVE_INDEXED_GZIP and file_like.endswith('gz')
+        # don't have indexed_gzip - auto -> False
+        if keep_file_open == 'auto' and not HAVE_INDEXED_GZIP:
+            return False
+        return keep_file_open
```

Comment:
We're dropping consideration of `keep_file_open == 'auto' and not file_like.endswith('gz')`. In this case, we will keep uncompressed files open, unnecessarily.

I would suggest making this final line:

```Python
return keep_file_open and file_like.endswith('gz')
```

Reply:
Aah - I left out the `endswith('.gz')` test intentionally, because if the user says `keep_file_open`, we should do just that, regardless of the file type, right?

But I've added the test back in, as `keep_file_open == 'auto'` and `not file_like.endswith('.gz')` should revert to `keep_file_open == False`. Commits coming in a few minutes!

Reply:
> because if the user says keep_file_open, we should do just that, regardless of the file type, right?

Right.

> But I've added the test back in, as keep_file_open == 'auto' and not file_like.endswith('.gz') should revert to keep_file_open == False

Yes. That's what I should have said. Good catch.